test/test_editor_loads_native.py

```diff
@@ -1,57 +1,80 @@
 from viscm.gui import Colormap, viscm_editor
 
-cms = {
-    "viscm/examples/sample_linear.jscm",
-    "viscm/examples/sample_diverging.jscm",
-    "viscm/examples/sample_diverging_continuous.jscm",
-}
-
-
-def test_editor_loads_native():
-    for k in cms:
-        with open(k) as f:
-            data = json.loads(f.read())
-        cm = Colormap(None, "CatmulClark", "CAM02-UCS")
-        cm.load(k)
-        viscm = viscm_editor(
-            uniform_space=cm.uniform_space,
-            cmtype=cm.cmtype,
-            method=cm.method,
-            **cm.params,
-        )
-        assert viscm.name == data["name"]
-
-        extensions = data["extensions"]["https://matplotlib.org/viscm"]
-        xp, yp, fixed = viscm.control_point_model.get_control_points()
-
-        assert extensions["fixed"] == fixed
-        assert len(extensions["xp"]) == len(xp)
-        assert len(extensions["yp"]) == len(yp)
-        assert len(xp) == len(yp)
-        for i in range(len(xp)):
-            assert extensions["xp"][i] == xp[i]
-            assert extensions["yp"][i] == yp[i]
-        assert extensions["min_Jp"] == viscm.min_Jp
-        assert extensions["max_Jp"] == viscm.max_Jp
-        assert extensions["filter_k"] == viscm.filter_k
-        assert extensions["cmtype"] == viscm.cmtype
-
-        colors = data["colors"]
-        colors = [
-            [int(c[i : i + 2], 16) / 256 for i in range(0, 6, 2)]
-            for c in [colors[i : i + 6] for i in range(0, len(colors), 6)]
-        ]
-        editor_colors = viscm.cmap_model.get_sRGB(num=256)[0].tolist()
-        for i in range(len(colors)):
-            for z in range(3):
-                assert colors[i][z] == np.rint(editor_colors[i][z] / 256)
+
+def approxeq(x, y, *, err=0.0001):
+    return abs(y - x) < err
+
+
+@pytest.mark.parametrize(
+    "colormap_file",
+    [
+        "viscm/examples/sample_linear.jscm",
+        "viscm/examples/sample_diverging.jscm",
+        "viscm/examples/sample_diverging_continuous.jscm",
+    ],
+)
+def test_editor_loads_native(colormap_file):
+    with open(colormap_file) as f:
+        data = json.loads(f.read())
+    cm = Colormap(None, "CatmulClark", "CAM02-UCS")
+    cm.load(colormap_file)
+    viscm = viscm_editor(
+        uniform_space=cm.uniform_space,
+        cmtype=cm.cmtype,
+        method=cm.method,
+        **cm.params,
+    )
+    assert viscm.name == data["name"]
+
+    extensions = data["extensions"]["https://matplotlib.org/viscm"]
+    xp, yp, fixed = viscm.control_point_model.get_control_points()
+
+    assert extensions["fixed"] == fixed
+    assert len(extensions["xp"]) == len(xp)
+    assert len(extensions["yp"]) == len(yp)
+    assert len(xp) == len(yp)
+    for i in range(len(xp)):
+        assert extensions["xp"][i] == xp[i]
+        assert extensions["yp"][i] == yp[i]
+    assert extensions["min_Jp"] == viscm.min_Jp
+    assert extensions["max_Jp"] == viscm.max_Jp
+    assert extensions["filter_k"] == viscm.cmap_model.filter_k
```

Comment:
This line was broken because of an interface change on the `viscm_editor` class since the test was originally developed (56255bcf65588eccf0862abb38680fd61a2ddbb7 in 2016, these tests have been broken for some time :) )